packages/openai/src/openai-transcription-model.ts

```diff
@@ -0,0 +1,208 @@
+const languageMap = {
+  afrikaans: 'af',
+  arabic: 'ar',
+  armenian: 'hy',
+  azerbaijani: 'az',
+  belarusian: 'be',
+  bosnian: 'bs',
+  bulgarian: 'bg',
+  catalan: 'ca',
+  chinese: 'zh',
+  croatian: 'hr',
+  czech: 'cs',
+  danish: 'da',
+  dutch: 'nl',
+  english: 'en',
+  estonian: 'et',
+  finnish: 'fi',
+  french: 'fr',
+  galician: 'gl',
+  german: 'de',
+  greek: 'el',
+  hebrew: 'he',
+  hindi: 'hi',
+  hungarian: 'hu',
+  icelandic: 'is',
+  indonesian: 'id',
+  italian: 'it',
+  japanese: 'ja',
+  kannada: 'kn',
+  kazakh: 'kk',
+  korean: 'ko',
+  latvian: 'lv',
+  lithuanian: 'lt',
+  macedonian: 'mk',
+  malay: 'ms',
+  marathi: 'mr',
+  maori: 'mi',
+  nepali: 'ne',
+  norwegian: 'no',
+  persian: 'fa',
+  polish: 'pl',
+  portuguese: 'pt',
+  romanian: 'ro',
+  russian: 'ru',
+  serbian: 'sr',
+  slovak: 'sk',
+  slovenian: 'sl',
+  spanish: 'es',
+  swahili: 'sw',
+  swedish: 'sv',
+  tagalog: 'tl',
+  tamil: 'ta',
+  thai: 'th',
+  turkish: 'tr',
+  ukrainian: 'uk',
+  urdu: 'ur',
+  vietnamese: 'vi',
+  welsh: 'cy',
+};
+
+export class OpenAITranscriptionModel implements TranscriptionModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: OpenAITranscriptionModelId,
+    private readonly config: OpenAITranscriptionModelConfig,
+  ) {}
+
+  async doGenerate({
+    audio,
+    providerOptions,
+    headers,
+    abortSignal,
+  }: Parameters<TranscriptionModelV1['doGenerate']>[0]): Promise<
+    Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>
+  > {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const formData = new FormData();
+
+    let blob: Blob | undefined;
+
+    if (audio instanceof Uint8Array) {
+      // Convert Uint8Array to Blob and then to File
+      blob = new Blob([audio]);
+    } else if (typeof audio === 'string') {
+      // Convert base64 string to Blob and then to File
+      const byteCharacters = atob(audio);
+      const byteNumbers = new Array(byteCharacters.length);
+      for (let i = 0; i < byteCharacters.length; i++) {
+        byteNumbers[i] = byteCharacters.charCodeAt(i);
+      }
+      const byteArray = new Uint8Array(byteNumbers);
+      blob = new Blob([byteArray]);
+    } else {
+      throw new Error(
+        'Invalid audio format. Must be Uint8Array or base64 string.',
+      );
+    }
+
+    formData.append('model', this.modelId);
+    formData.append(
+      'file',
+      new File([blob], 'audio.wav', { type: 'audio/wav' }),
+    );
+
+    // Add any additional provider options
+    if (providerOptions?.openai) {
+      for (const [key, value] of Object.entries(providerOptions.openai)) {
+        if (key in providerOptionsMapping) {
+          const newKey =
+            providerOptionsMapping[key as keyof typeof providerOptionsMapping];
+
+          formData.append(newKey, String(value));
+        }
+      }
+    }
```

Comment:
Please use the validation approach for provider options that we use elsewhere (so we get a type for type checking and can throw errors before the request). See https://github.com/vercel/ai/blob/main/packages/groq/src/groq-chat-language-model.ts#L105 for an example

Reply:
Added validation for both the raw options and provider options + extended the `providerOptions` for better type safety.